### DISCUSSION TITLE
human task assignment

### DISCUSSION THREAD

**Discussion by bsdgithub** [flowsforapex/apex-flowsforapex#719] [Q&A]
Hello,
I have a simple flow with two lanes. Also, the users are working from different locations, and I have groups like "Location1_ACCOUNTING_MAKER","Location1_ACCOUNTING_CHECKER", "Location2_ACCOUNTING_MAKER", etc.
Is there a suggestion how should I deal with task assignment? Maker should be the one that will start the process and on the other lane checker will finalize it.
Thanks,

**Comment by rallen2010** [comment]
Flows for APEX 23.1 (and later) contains mechanisms to help you with Task Assignment that allow you to create very flexible, data driven rules.

Start with BPMN Lanes.   She lanes map to roles or groups that you already have defined in your system - but others often do not.  The expense claim example where a claim must be signed by your manager, and the claim goes returned to you if it is rejected - are examples where lanes do not nicely map to group assignment.   Your expenses go to YOUR manager, not every manager in the company, and your returned claim goes to YOU - not every employee in the company.  So those tasks need to be assigned to the process originator and to the person who is the originator's manager (database lookup) respectively, at TASK level - not by assigning all tasks in the lane to EMPLOYEES and MANAGERS roles.  

Once approved, they might all flow to the FINANCE - ACCOUNTS PAYABLE - EXPENSES clerk - which IS a role in your system.   In the BPMN Modeler, If you select a LANE object, in the properties panel > Assignment section, you can define that a lane maps to an APEX role, and specify the role name.   That then becomes the default task assignment for all tasks in that lane.   Note that the default can be overridden if you explicitly define assignment rules for any task in the lane.

And at task level, task assignment is calculated when a task becomes current.  You have a range of ways to set the user or group that it should be assigned to - from a static assignment, the value in a process variable, or a SQL query of plsql expression / function body that can evaluate all your business rules and current database data.  So if you can define your riule, you should be able to enforce it.
